log/validator/validator.go

```diff
@@ -0,0 +1,171 @@
+package validator
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/nxadm/tail"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/letsencrypt/boulder/log"
+)
+
+var errInvalidChecksum = errors.New("invalid checksum length")
+
+type Validator struct {
+	tailers []*tail.Tail
+
+	lineCounter *prometheus.CounterVec
+	log         log.Logger
+}
+
+func New(logger log.Logger, stats prometheus.Registerer) *Validator {
+	lineCounter := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "log_lines",
+		Help: "A counter of log lines processed, with status",
+	}, []string{"filename", "status"})
+	stats.MustRegister(lineCounter)
+
+	return &Validator{log: logger, lineCounter: lineCounter}
+}
+
+// TailValidateFile takes a filename, and starts tailing it.
+// An error is returned if the file couldn't be opened.
+func (v *Validator) TailValidateFile(filename string) error {
+	t, err := tail.TailFile(filename, tail.Config{
+		ReOpen:        true,
+		MustExist:     false, // sometimes files won't exist, so we must tolerate that
+		Follow:        true,
+		Logger:        tailLogger{v.log},
+		CompleteLines: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		// Emit no more than 1 error line per second. This prevents consuming large
+		// amounts of disk space in case there is problem that causes all log lines to
+		// be invalid.
+		outputLimiter := time.NewTicker(time.Second)
```

Comment:
You'll should call `outputLimiter.Stop()` to stop this Ticker once it's no longer needed. You might just be able to do a defer for that.